beep-beep/jobs.md

```diff
@@ -0,0 +1,695 @@
+plugin. To enable it add `jobs` section to your configuration.
+
+For example, in this way, you can configure both the client and server parts to
+work with RabbitMQ.
+
+```yaml
+#
+# RPC is required for tasks dispatching (client)
+#
+rpc:
+  listen: tcp://127.0.0.1:6001
+
+#
+# This section configures the task consumer (server)
+#
+server:
+  command: php consumer.php
+  relay: pipes
+
+#
+# AMQP connection settings (for example, RabbitMQ)
+#
+amqp:
+    addr: amqp://guest:guest@localhost:5672
+
+#
+# In this section, the jobs themselves are configured
+#
+jobs:
+  consume: [ "test" ]   # List of RoadRunner queues that can be processed by 
+                        # the consumer specified in the "server" section.
+  pipelines:
+    test:               # RoadRunner queue identifier
+      driver: amqp      # - Queue driver name
+      queue: test       # - Internal (driver's) queue identifier
+```
+
+- The `rpc` section is responsible for client settings. It is at this address
+  that we will connect, *dispatching tasks* to the queue.
+
+- The `server` section is responsible for configuring the server. Previously, we
+  have already met with its description when setting up the [PHP Worker](/php/worker.md).
+
+- The `amqp` section contains settings for connecting to RabbitMQ. As the name
+  implies, the settings are specific only for the RabbitMQ driver and *may not*
+  be available if you use some other driver.
+
+- And finally, the `jobs` section is responsible for the work of the queues
+  themselves. It contains information on how the RoadRunner should work with
+  connections to drivers, what can be handled by the consumer, and other
+  queue-specific settings.
+
+### Common Configuration
+
+Let's now focus on the common settings of the queue server. In full, it may 
+look like this:
+
+```yaml
+jobs:
+  num_pollers: 64
+  pipeline_size: 100000
+  pool:
+    num_workers: 10
+    max_jobs: 0
+    allocate_timeout: 60s
+    destroy_timeout: 60s
+  consume: [ "queue-name" ]
+  pipelines:
+    queue-name:
+      driver: # "[DRIVER_NAME]"
+      # And driver-specific configuration below...
+```
+
+Above is a complete list of all possible common Jobs settings. Let's now figure
+out what they are responsible for.
+
+- `num_pollers` - TODO
+- `pipeline_size` - TODO
+- `pool.num_workers` - TODO
+- `pool.max_jobs` - TODO
+- `pool.allocate_timeout` - TODO
+- `pool.destroy_timeout` - TODO
+- `consume` - Contains an array of the names of all queues specified in the
+  `"pipelines"` section, which should be processed by the concierge specified in
+   the global `"server"` section (see the [PHP worker's settings](/php/worker.md)).
+- `pipelines` - This section contains a list of all queues declared in the
+  RoadRunner. The key is a unique *queue identifier*, and the value is an object
+  from the settings specific to each driver (we will talk about it later).
+
+### Ephemeral (Memory) Driver
+
+TODO
+
```

Comment:
```json
      driver: ephemeral
      priority: 10
      prefetch: 10000
```

Where:
1. `driver, priority` - same.
2. `prefetch` - a local buffer between the PQ and driver. If the PQ size is set to 100 and `prefetch` to 100000, you'll be able to push up to `prefetch` number of jobs even if PQ is full.
3. Internal goroutines limiter - 1000. You won't be allowed to have at the same time more than 1000 jobs with delay.